Mathlib/RingTheory/IntegralRestrict.lean

```diff
@@ -0,0 +1,129 @@
+/-
+Copyright (c) 2023 Andrew Yang, Patrick Lutz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.RingTheory.IntegrallyClosed
+import Mathlib.RingTheory.Norm
+import Mathlib.RingTheory.DedekindDomain.IntegralClosure
+/-!
+# Restriction of various maps between fields to integrally closed subrings.
+
+## Main definition
+- `galRestrict`: The restriction `Aut(L/K) → Aut(B/A)` as an `MulEquiv` in an ALKB setup.
+
+## TODO
+Define the restriction of norms and traces.
+
+-/
+open BigOperators nonZeroDivisors
+
+variable (A K L B : Type*) [CommRing A] [CommRing B] [Algebra A B] [Field K] [Field L]
+    [Algebra A K] [IsFractionRing A K] [Algebra B L]
+    [Algebra K L] [Algebra A L] [IsScalarTower A B L] [IsScalarTower A K L]
+    [IsIntegralClosure B A L] [FiniteDimensional K L]
+
+/-- The restriction `Aut(L/K) → Aut(B/A)` in an ALKB setup.
+Also see `galRestrict` for the `AlgEquiv` version. -/
+noncomputable
+def galRestrictHom : (L →ₐ[K] L) ≃* (B →ₐ[A] B) where
```

Comment:
This is a rather long definition. Maybe a good idea is to split it, defining the function and proving the various properties, and then packaging everything here (you can make everything `private` if you want).